awx/ui/src/components/ResourceAccessList/ResourceAccessList.js

```diff
@@ -56,31 +56,29 @@ function ResourceAccessList({ apiModel, resource }) {
 
       let orgRoles;
       if (location.pathname.includes('/organizations')) {
-        const {
-          data: { results: roles },
-        } = await RolesAPI.read({ content_type__isnull: true });
-        const sysAdmin = roles.filter(
-          (role) => role.name === 'System Administrator'
-        );
-        const sysAud = roles.filter((role) => {
-          let auditor;
-          if (role.name === 'System Auditor') {
-            auditor = role.id;
-          }
-          return auditor;
-        });
+        const [
+          {
+            data: { results: systemAdmin },
+          },
+          {
+            data: { results: systemAuditor },
+          },
+        ] = await Promise.all([
+          RolesAPI.read({ singleton_name: 'system_administrator' }),
```

Comment:
Nice! I didn't realize we could query this directly - this lifts some of the complexity from the UI code.

Reply:
That was @AlanCoding magic. 🧙